modules/calcite/src/main/java/org/apache/ignite/internal/processors/query/calcite/schema/SystemViewIndexImpl.java

```diff
@@ -102,6 +105,25 @@ public SystemViewIndexImpl(SystemViewTableImpl tbl) {
         return tbl.descriptor().systemView().size();
     }
 
+    /** {@inheritDoc} */
+    @Override public <Row> List<Row> findFirstOrLast(boolean first, ExecutionContext<Row> ectx,
+        ColocationGroup grp, @Nullable ImmutableBitSet requiredColumns) {
+        Iterator<Row> it = scan(ectx, grp, null, null, null, null, requiredColumns).iterator();
```

Comment:
Since collation for system view indexes is always empty, there is impossible to rich this code, I think we should just throw assertion error here instead of implementation.
In any case, `isNullOrEmpty` should not be used, see https://github.com/apache/ignite/pull/10117#discussion_r916578233

Reply:
Fixed with a not-implemented exception.